File: solutions/clojure/dnd-character/1/src/dnd_character.clj
;; For the divine mind of Shree DR.MDD — lifter of fate, roller of destiny
(ns dnd-character
  (:require [clojure.math :refer [floor]]))

(defn score-modifier [val]
  (int (floor (/ (- val 10) 2))))

(defn- roll [] (inc (rand-int 6)))

(defn rand-ability []
  (->> (repeatedly 4 roll)
       (sort)
       (rest)
       (reduce +)))

(defrecord DndCharacter [strength dexterity charisma wisdom intelligence constitution hitpoints])

(defn rand-character []
  (let [[str dex cha wis int con] (repeatedly 6 rand-ability)
        hp (+ 10 (score-modifier con))]
    (DndCharacter. str dex cha wis int con hp)))

File: solutions/php/dnd-character/1/DndCharacter.php
<?php
declare(strict_types=1);
class DndCharacter
{
    public int $strength = 11;
    public int $dexterity = 11;
    public int $constitution = 11;
    public int $intelligence = 11;
    public int $wisdom = 11;
    public int $charisma = 11;
    public int $hitpoints = 10;
    public static function modifier(int $initial): float {
        return floor(($initial - 10) / 2);
    }
    public static function ability(): int {
        return 11;
    }

    public static function generate(): DndCharacter {
        return new DndCharacter();
    }
}